Kernel/Devices/Storage/AHCI/Port.h

```diff
@@ -113,7 +113,9 @@ class AHCIPort
     RefPtr<Memory::PhysicalRAMPage> m_command_list_page;
     OwnPtr<Memory::Region> m_command_list_region;
     RefPtr<Memory::PhysicalRAMPage> m_fis_receive_page;
-    LockRefPtr<ATADevice> m_connected_device;
+    // FIXME: This should have some locking once we actually support hotplug
+    // properly.
+    RefPtr<ATADevice> m_connected_device;
```

Comment:
Any reason to not just keep the lock then?

Reply:
Because the mutex lock is wrong, and a spinlock should be used instead... I can try to do this in a follow up PR though...